package.json

```diff
@@ -21,7 +21,7 @@
     "record-build-info": "node ./bin/record-build-info",
     "lint": "eslint . --cache --max-warnings 0",
     "typecheck": "tsc && tsc -p integration_tests",
-    "test": "jest",
+    "test": "jest --detectOpenHandles",
```

Comment:
this is a big hack.  Turning this on then forces `jest` to run single threaded which seems to fix the issue.  I've been through codebase and can't see any missing `await`s so not sure what is going on.  Detecting open handles doesn't find any anywhere :shrug: